to.etc.domui/src/main/java/to/etc/domui/dom/html/Page.java

```diff
@@ -1380,4 +1390,31 @@ public void execute(@NonNull IExecute code) {
 				page.addDelayedExecution(code);
 		}
 	}
+
+	/**
+	 * Checks if page can be left.
+	 * @return
+	 */
+	public boolean canLeaveCurrentPage(boolean handleNavigationWhenModified) {
+		if(m_forceLeavePage) {
+			return true;
+		}
+		if(m_rootContent instanceof IPageWithNavigationCheck) {
+			IPageWithNavigationCheck pageWithNavigationCheck = (IPageWithNavigationCheck) m_rootContent;
+			boolean hasModification = pageWithNavigationCheck.hasModification();
+			if(hasModification && handleNavigationWhenModified) {
+				pageWithNavigationCheck.handleNavigationOnModified();
+			}
+			return !hasModification;
+		}else {
+			return true;
+		}
+	}
+
+	/**
+	 * Forces page leave flag, it would ignore page leave check on url page level.
+	 */
+	public void setForceLeavePage() {
```

Comment:
This is an obscure way to handle this: no one would be able to really find this as part of that interface.

What is a better (and self explanatory) way of doing this is by having the handler of the "the page is modified" action receive a parameter that contains the method to be called. That might also be the place where to store the missing UIGoto information..

Reply:
removed it, not needed now